<launch>
 <param name="/use_sim_time" value="true" />
 <param name="root_frame" value="map" />
 <param name="child_frame" value="UGV_odom" />
 
 <param name="latitude_reference" value="45.477669461666665" />
 <param name="longitude_reference" value="9.22674018" />
 <param name="altitude_reference" value="169.039" />

<node pkg="second_project" name="lidar_remap" type="lidar_remap"/>
<node pkg="second_project" name="set_goals" type="set_goals" output="screen"/>

<!-- Static Transform Publisher between map and odom -->
<!--  <node pkg="tf2_ros" type="static_transform_publisher" name="map_to_odom_frame" args="0 0 0 0 0 0 map odom_frame" /> -->

 <!-- <node pkg="tf" type="static_transform_publisher" name="odom_to_rslidar" args="0 0 0 0 0 0 /ugv/odom rslidar 100"/> -->

 <node pkg="slam_toolbox" type="async_slam_toolbox_node" name="slam_toolbox" output="screen">
    <rosparam command="load" file="$(find second_project)/cfg/mapper_params_online_async.yaml" />
    <!-- <remap from="pointcloud_scan" to="base_scan" /> -->
  </node>

<node name="pointcloud_to_laserscan" pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan_node" output="screen">
       <remap from="cloud_in" to="pointcloud_remapped" />
       <remap from="scan" to="pointcloud_scan" />
       
       <!-- Parameters -->
       <param name="target_frame" value="UGV_odom"/>
       <param name="min_height" value="0.1"/>
       <param name="max_height" value="10.0"/>
       <param name="angle_min" value="-3.14159"/>
       <param name="angle_max" value="3.14159"/>
       <param name="angle_increment" value="0.001745"/>
       <param name="range_min" value="0.45"/>
       <param name="range_max" value="10.0"/>
       <param name="use_inf" value="true"/>
</node>

<group>
    <node pkg="second_project" name="odom_to_tf" type="odom_to_tf">
        <param name="root_frame" value="map"/>
        <param name="child_frame" value="UGV_odom"/>
    </node>
</group>
    
<arg name="rvizconfig" default="$(find second_project)/rviz/robot_mapping.rviz" />
<node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" />

</launch>
